plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -99,6 +108,8 @@ public class IcebergSplitSource
 {
     private static final ConnectorSplitBatch EMPTY_BATCH = new ConnectorSplitBatch(ImmutableList.of(), false);
     private static final ConnectorSplitBatch NO_MORE_SPLITS_BATCH = new ConnectorSplitBatch(ImmutableList.of(), true);
+    // It's a static value instead of a config, in order to protect users from setting this too high accidentally and causing coordinator memory issues
+    private static final int MAX_OUTSTANDING_SPLITS = 100;
```

Comment:
100 outstanding splits sounds like a rather conservative value.

in fact, BufferingSplitSource by default uses buffer size of 100.
I don't know why we need BufferingSplitSource while we also have buffering inside the connector (and even once attempted to remove it https://github.com/trinodb/trino/pull/7394 but failed). Assuming we need both, Iceberg should have a bigger buffer size, and it should be configurable, jusjt like BufferingSplitSource is.

Reply:
Thanks @findepi. Other connector implementations, like [DeltaLakeConfig](https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java#L57) and [HudiConfig](https://github.com/trinodb/trino/blob/master/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java#L49) set the default to 1000 for this value (and [Hive](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java#L77) even higher, to 3000). Initially, I also made this configurable and set the default to 1000 to be consistent with the other connectors. 

@raunaqmorarka had a concern that users might accidentally set this value too high and increase the memory consumption. Perhaps this is a relatively unlikely scenario if the other three connectors have not experienced this issue? Let me know if I should make this configurable again and set to 1000 default